pandas-stubs/core/indexes/interval.pyi

```diff
@@ -11,80 +13,210 @@ import numpy as np
+        dtype: IntervalDtype | None = ...,
+    ) -> IntervalIndex[Interval[int]]: ...
+    @overload
+    @classmethod
+    def from_arrays(
+        cls,
+        left: _EdgesFloat,
+        right: _EdgesFloat,
+        closed: IntervalClosedType = ...,
+        name: Hashable = ...,
+        copy: bool = ...,
+        dtype: IntervalDtype | None = ...,
+    ) -> IntervalIndex[Interval[float]]: ...
+    @overload
     @classmethod
     def from_arrays(
         cls,
-        left: _Edges,
-        right: _Edges,
+        left: _EdgesTimestamp,
+        right: _EdgesTimestamp,
         closed: IntervalClosedType = ...,
         name: Hashable = ...,
         copy: bool = ...,
         dtype: IntervalDtype | None = ...,
-    ) -> IntervalIndex: ...
+    ) -> IntervalIndex[Interval[pd.Timestamp]]: ...
+    @overload
+    @classmethod
+    def from_arrays(
+        cls,
+        left: _EdgesTimedelta,
+        right: _EdgesTimedelta,
+        closed: IntervalClosedType = ...,
+        name: Hashable = ...,
+        copy: bool = ...,
+        dtype: IntervalDtype | None = ...,
+    ) -> IntervalIndex[Interval[pd.Timedelta]]: ...
+    @overload
     @classmethod
     def from_tuples(
         cls,
-        data: Sequence[tuple[int, int]]
-        | Sequence[tuple[float, float]]
-        | Sequence[tuple[DatetimeLike, DatetimeLike]]
-        | npt.NDArray,
+        data: Sequence[tuple[int, int]],
         closed: IntervalClosedType = ...,
         name: Hashable = ...,
         copy: bool = ...,
         dtype: IntervalDtype | None = ...,
-    ) -> IntervalIndex: ...
+    ) -> IntervalIndex[pd.Interval[int]]: ...
+    # Ignore misc here due to intentional overlap between int and float
+    @overload
+    @classmethod
+    def from_tuples(
+        cls,
+        data: Sequence[tuple[float, float]],
+        closed: IntervalClosedType = ...,
+        name: Hashable = ...,
+        copy: bool = ...,
+        dtype: IntervalDtype | None = ...,
+    ) -> IntervalIndex[pd.Interval[float]]: ...
+    @overload
+    @classmethod
+    def from_tuples(
+        cls,
+        data: Sequence[
+            tuple[pd.Timestamp, pd.Timestamp]
+            | tuple[dt.datetime, dt.datetime]
+            | tuple[np.datetime64, np.datetime64]
+        ],
+        closed: IntervalClosedType = ...,
+        name: Hashable = ...,
+        copy: bool = ...,
+        dtype: IntervalDtype | None = ...,
+    ) -> IntervalIndex[pd.Interval[pd.Timestamp]]: ...
+    @overload
+    @classmethod
+    def from_tuples(
+        cls,
+        data: Sequence[
+            tuple[pd.Timedelta, pd.Timedelta]
+            | tuple[dt.timedelta, dt.timedelta]
+            | tuple[np.timedelta64, np.timedelta64]
+        ],
+        closed: IntervalClosedType = ...,
+        name: Hashable = ...,
+        copy: bool = ...,
+        dtype: IntervalDtype | None = ...,
+    ) -> IntervalIndex[pd.Interval[pd.Timedelta]]: ...
     def __contains__(self, key: Any) -> bool: ...
```

Comment:
I think we could probably do `def __contains__(self: key: IntervalT) -> bool: ...` here or something like that.  Might have to deal with the `int/float` dichotomy with some overloads.

Reply:
Updated this